cmake/cmaize/targets/target.cmake

```
@@ -0,0 +1,239 @@
+    #    analog, ``Target`` really shouldn't be instantiated aside from
+    #    testing purposes. Instead, create a child with a concrete target
+    #    analog and instantiate that.
+    #
+    # :param self: Target object constructed.
+    # :type self: Target
+    # :param tgt_name: Name of the target. This should not duplicate any other
+    #                  target name already in scope.
+    # :type tgt_name: str
+    # :returns: ``self`` will be set to the newly constructed ``Target``
+    #           object.
+    # :rtype: Target
+    #]]
+    cpp_constructor(CTOR Target str)
+    function("${CTOR}" self tgt_name)
+
+        Target(SET "${self}" _name "${tgt_name}")
+
+    endfunction()
+
+    #[[[
+    # Creates a ``Target`` object based on an existing CMake target.
+    #
+    # :param self: Target object constructed.
+    # :type self: Target
+    # :param tgt_name: Name of the existing target.
+    # :type tgt_name: str
+    #
+    # :returns: ``self`` will be set to the newly constructed ``Target``
+    #           object.
+    # :rtype: Target
+    #]]
+    cpp_constructor(CTOR Target target)
+    function("${CTOR}" self tgt_name)
+
+        Target(SET "${self}" _name "${tgt_name}")
+
+    endfunction()
+
+    #[[[
+    # Get the CMake target name that the ``Target`` class represents.
+    # 
+    # :param self: Target object
+    # :type self: Target
+    # :param return_target: Name of the CMake target.
+    # :type return_target: str
+    #
+    # :returns: Sets ``return_target`` to the name of the CMake target
+    #           represented by this class.
+    # :rtype: str
+    #]]
+    cpp_member(target Target str)
+    function("${target}" self return_target)
+
+        Target(GET "${self}" "${return_target}" _name)
+
+        cpp_return("${return_target}")
+
+    endfunction()
+
+    #[[[
+    # Check if the target has the requested property.
+    #
+    # .. note::
+    #    
+    #    The signature is currently ``cpp_member(has_property Target desc str)``
+    #    instead of ``cpp_member(has_property Target bool str)`` due to a bug
+    #    in CMakePPLang that throws an overload error when a non-boolean is
+    #    passed into the ``bool`` parameter. Since this is a return value, we
+    #    always pass a non-boolean in and will always trigger the error. By
+    #    setting it to ``desc`` type instead, it will accept non-booleans
+    #    and the user will have to trust that a boolean string is being
+    #    returned from the docs.
+    # 
+    # :param self: Target object
+    # :type self: Target
+    # :param has_property: Return variable for if the target has the property.
+    # :type has_property: bool
+    # :param property_name: Name of the property to check for.
+    # :type property_name: str
+    #
+    # :returns: Sets ``has_property`` according to if the target has the
+    #           requested property (True) or not (False).
+    # :rtype: bool
```

```suggestion
    # :rtype: bool*
```
FYI https://cmakepp.github.io/CMakePPLang/features/types.html#pointers is how I recommend documenting the return type.

---

Would you recommend using the pointer typing for all return variables?

---

yep